### PR TITLE
Opimize mesos state store scheduler handling

### PR DESCRIPTION
**Pull request by orlandohohmeier**
---
⚠️  ~~_This branch depends on changes that will be introduced with #1566_~~

---

Improve `MesosStateStore` scheduler task handling by introducing a new method to retrieve the list of scheduler tasks and simplifying the existing methods. These changes will reduce the number of `framework`/`task` list iterations. 

/cc @MatApple

### CODE REVIEW

File: src/js/stores/__tests__/MesosStateStore-test.js [hunk 1]
```diff
@@ -180,6 +180,52 @@ describe('MesosStateStore', function () {
...
+                id: 1,
+                labels: [{key: 'DCOS_PACKAGE_FRAMEWORK_NAME', value: 'foo'}]
+              },
+              {
+                id: 2
+              },
+              {
+                id: 3,
+                labels: [{key: 'DCOS_PACKAGE_FRAMEWORK_NAME', value: 'bar'}]
+              }
```
Comment (mlunoe) [inline src/js/stores/__tests__/MesosStateStore-test.js]: Could we also add some tasks that will not be included, to test out that scenario as well?

File: src/js/stores/MesosStateStore.js [hunk 2]
```diff
@@ -229,36 +220,30 @@ class MesosStateStore extends GetSetBaseStore {
...
     });
+  }
 
-    if (!framework || !framework.tasks) {
-      return null;
-    }
-
-    let result = framework.tasks.find(function (task) {
-      let labels = task.labels;
+  getSchedulerTaskFromServiceName(serviceName) {
```
Comment (mlunoe) [inline src/js/stores/MesosStateStore.js]: Same thing for the existing tests for this function

**Comment by orlandohohmeier** [comment]
@mlunoe addressed your comments...

**Comment by nLight** [comment]
✨ 💯

**Pull request merged**